Handle notifications with large number of tickets and use `docker compose` syntax

dav3r opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR resolves a situation where a Mongo query can run out of memory and fail if a notification contains a large number of tickets.

In addition, https://github.com/cisagov/cyhy-reports/pull/77/commits/5485ed9f93fc36d2ec1cade471fb302cc9839988 updates `create_send_notifications.py` to use the newer `docker compose` syntax instead of the older style `docker-compose` syntax.  The `docker compose` syntax is the preferred (and only correct) syntax after the changes in https://github.com/cisagov/ansible-role-docker/pull/60.

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

After our recent switch to CVSS v3 (see https://github.com/cisagov/cyhy-core/pull/69 and https://github.com/cisagov/cyhy-reports/pull/76), our daily notification process failed because one of the organizations had a very large number of tickets to process for their notification.  The process failed with this error:

```
Jun 10 08:15:55 reporter cyhy-notifications: 2022-06-10 08:15:55,705 INFO <ORG_ID> - Starting to create notification PDF
Jun 10 08:15:56 reporter cyhy-notifications: Traceback (most recent call last):
Jun 10 08:15:56 reporter cyhy-notifications:   File "./create_send_notifications.py", line 184, in <module>
Jun 10 08:15:56 reporter cyhy-notifications:     sys.exit(main())
Jun 10 08:15:56 reporter cyhy-notifications:   File "./create_send_notifications.py", line 115, in main
Jun 10 08:15:56 reporter cyhy-notifications:     num_pdfs_created = generate_notification_pdfs(db, cyhy_org_ids, master_report_key)
Jun 10 08:15:56 reporter cyhy-notifications:   File "./create_send_notifications.py", line 72, in generate_notification_pdfs
Jun 10 08:15:56 reporter cyhy-notifications:     was_encrypted, results = generator.generate_notification()
Jun 10 08:15:56 reporter cyhy-notifications:   File "/usr/local/lib/python2.7/dist-packages/cyhy_report/cyhy_notification/generate_notification.py", line 162, in generate_notification
Jun 10 08:15:56 reporter cyhy-notifications:     self.__run_queries()
Jun 10 08:15:56 reporter cyhy-notifications:   File "/usr/local/lib/python2.7/dist-packages/cyhy_report/cyhy_notification/generate_notification.py", line 361, in __run_queries
Jun 10 08:15:56 reporter cyhy-notifications:     self.__results["tickets"] = self.__load_tickets(ticket_ids)
Jun 10 08:15:56 reporter cyhy-notifications:   File "/usr/local/lib/python2.7/dist-packages/cyhy_report/cyhy_notification/generate_notification.py", line 271, in __load_tickets
Jun 10 08:15:56 reporter cyhy-notifications:     ("details.name", 1),
Jun 10 08:15:56 reporter cyhy-notifications:   File "/usr/local/lib/python2.7/dist-packages/mongokit/cursor.py", line 42, in next
Jun 10 08:15:56 reporter cyhy-notifications:     if len(self.__data) or self._refresh():
Jun 10 08:15:56 reporter cyhy-notifications:   File "/usr/local/lib/python2.7/dist-packages/pymongo/cursor.py", line 1095, in _refresh
Jun 10 08:15:56 reporter cyhy-notifications:     self.__codec_options.uuid_representation))
Jun 10 08:15:56 reporter cyhy-notifications:   File "/usr/local/lib/python2.7/dist-packages/pymongo/cursor.py", line 1037, in __send_message
Jun 10 08:15:56 reporter cyhy-notifications:     self.__compile_re)
Jun 10 08:15:56 reporter cyhy-notifications:   File "/usr/local/lib/python2.7/dist-packages/pymongo/helpers.py", line 144, in _unpack_response
Jun 10 08:15:56 reporter cyhy-notifications:     error_object)
Jun 10 08:15:56 reporter cyhy-notifications: pymongo.errors.OperationFailure: database error: Executor error during OP_QUERY find :: caused by :: errmsg: "Sort operation used more than the maximum 33554432 bytes of RAM. Add an index, or specify a smaller limit."
```

I first attempted to fix this by adding a new index to support this sort, but I was unable to get the sort to use that index.  So instead, I worked around this issue by switching to a Mongo `aggregate` query, which allowed me to use the [`allowDiskUse` option](https://www.mongodb.com/docs/v3.6/reference/method/db.collection.aggregate/#method-aggregate-allowdiskuse) to circumvent the memory limitation.
 
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

To test these changes, I first verified that I could manually create a notification for the organization that failed with the memory error, and that was successful.  I then created a fresh notification for an org that had previously (this morning) successfully generated a notification, and I compared those two docs to ensure that they were identical (they were).

Finally, I executed the entire daily notification process, which ran without any errors.

<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

dav3r committed:
Switch from a find to an aggregation query

A regular find/sort query can exceed the memory limit of MongoDB if there are a large number of tickets included in a notification.

dav3r committed:
Update code to handle side effects of aggregation

Now that we are not using a regular find/sort query, our result set no longer contains TicketDocs, hence these changes.

felddy reviewed (approved):
`allowDiskUse=Booosh!`
Fine work @dav3r 💯

jsf9k reviewed (approved):
Powerful work as always, @dav3r.

mcdonnnj reviewed (approved):
🕵️‍♂️ :shipit:

dav3r committed:
Use "docker compose" syntax instead of "docker-compose"

The "docker compose" syntax is the preferred (and only correct) syntax after the changes in cisagov/ansible-role-docker#60.

jsf9k reviewed (approved):
_Approval intensifies_

felddy reviewed (approved):
- [x] Changes are limited to a single goal - *eschew scope creep!* 
🤣 

💪

mcdonnnj reviewed (approved):
🐳
